lib/web/apiserver.go

```diff
@@ -1036,7 +1036,11 @@ func (h *Handler) bindDefaultEndpoints() {
 	h.GET("/webapi/scripts/integrations/configure/eks-iam.sh", h.WithLimiter(h.awsOIDCConfigureEKSIAM))
 	h.GET("/webapi/scripts/integrations/configure/access-graph-cloud-sync-iam.sh", h.WithLimiter(h.accessGraphCloudSyncOIDC))
 	h.GET("/webapi/scripts/integrations/configure/aws-app-access-iam.sh", h.WithLimiter(h.awsOIDCConfigureAWSAppAccessIAM))
+	// TODO(kimlisa): DELETE IN 19.0 - Replaced by /v2 equivalent endpoint
 	h.POST("/webapi/sites/:site/integrations/aws-oidc/:name/aws-app-access", h.WithClusterAuth(h.awsOIDCCreateAWSAppAccess))
+	// v2 endpoint introduces "labels" field
+	// MUST delete with related code found in web/packages/teleport/src/services/integrations/integrations.ts(createAwsAppAccess)
+	h.POST("/v2/webapi/sites/:site/integrations/aws-oidc/:name/aws-app-access", h.WithClusterAuth(h.awsOIDCCreateAWSAppAccess))
```

Comment:
Is it needed/desired to make it v2 at all? It seems to be 100% backward compatible.

Reply:
if a new web UI hits an old proxy, it will throw out any `labels` data sent in a request and not add them to the resource, so we had to make a new endpoint that could handle that field.

Reply:
in addition, v2 was created so that the frontend can detect 404 not found failure, and tell the user that all proxy needs upgrading (instead of silently succeeding and the labels are nowhere to be found)